dbms/src/AggregateFunctions/AggregateFunctionForEach.h

```diff
@@ -173,20 +188,45 @@ class AggregateFunctionForEach final
                     ErrorCodes::SIZES_OF_ARRAYS_DOESNT_MATCH);
         }
 
-        AggregateFunctionForEachData & state = ensureAggregateData(place, end - begin, *arena);
+        AggregateFunctionForEachData & state = ensureAggregateData(place, end - begin, arena);
 
         char * nested_state = state.array_of_aggregate_datas;
         for (size_t i = begin; i < end; ++i)
         {
-            nested_func->add(nested_state, nested, i, arena);
+            if constexpr (is_add)
+                nested_func->add(nested_state, nested, i, arena);
+            else
+                nested_func->decrease(nested_state, nested, i, arena);
```

Comment:
why not just call `nested_func->addOrDecrease<is_add>()`?

Reply:
> why not just call `nested_func->addOrDecrease<is_add>()`?

Because `addOrDecrease<>()` is not an interface in `IAggregateFunction`.